main/src/main/resources/org/clulab/numeric/date-ranges.yml

```diff
@@ -85,6 +85,24 @@
   pattern: |
     /(?i)until|through/ @date1:Date
 
+- name: date-range-9
+  priority: ${rulepriority}
+  label: DateRange
+  type: token
+  example: "First week of May"
+  action: mkDateRangeMentionWithWeek
+  pattern: |
+    (?<week> /(?i)(first|1st|second|2nd|third|3rd|fourth|4th|last)/ /(?i)week/) /(?i)of/ @month:PossibleMonth
```

Comment:
This would match "first week of ..." no?

Reply:
@kwalcock, the pattern `?<week>` is used to label the argument (_week_) here. It is basically doing the same job as  `@arg` (e.g., `@month`). The reason I used `?<week>` instead of `@week` is because the pattern I'm extracting here has not been extracted else where. (IIRC, `@arg` can only be used with already extracted mentions)

Reply:
@MihaiSurdeanu, you are right. It will match patterns such as "first week", "1st week", "second week", "last week", and etc and label them as an argument  _week_.